Timeout while loading submissions

Example:

```[22-May-14 20:36:34] [ERROR] - Unable to calculate timeout location
[22-May-14 20:36:34] [ERROR] - Timeout after 1000ms @ DefaultDispatcher-worker-2
java.lang.Exception
        at org.sourcegrade.jagr.core.executor.TimeoutHandler.getTimeoutLocation(TimeoutHandler.kt:91)
        at org.sourcegrade.jagr.core.executor.TimeoutHandler.checkTimeout(TimeoutHandler.kt:74)
        at h01.utils.BytecodeTransformations.getWriterFlags(BytecodeTransformations.java)
        at org.sourcegrade.jagr.core.transformer.TransformerApplierKt.transform(TransformerApplier.kt:91)
        at org.sourcegrade.jagr.core.transformer.TransformerApplierKt.transform(TransformerApplier.kt:85)
        at org.sourcegrade.jagr.core.transformer.MultiTransformerApplierImpl.transform(TransformerApplier.kt:59)
        at org.sourcegrade.jagr.core.transformer.TransformerApplierKt.createApplier$lambda-1(TransformerApplier.kt:79)
        at org.sourcegrade.jagr.core.transformer.TransformerApplierKt.plus$lambda-0(TransformerApplier.kt:40)
        at org.sourcegrade.jagr.core.executor.CompiledBatchFactoryImpl$compile$2.invokeSuspend(BatchCompilation.kt:155)
        at org.sourcegrade.jagr.core.executor.CompiledBatchFactoryImpl$compile$2.invoke(BatchCompilation.kt)
        at org.sourcegrade.jagr.core.executor.CompiledBatchFactoryImpl$compile$2.invoke(BatchCompilation.kt)
        at org.sourcegrade.jagr.core.ParallelKt$parallel$deferreds$1$1.invokeSuspend(Parallel.kt:73)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:749)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
[22-May-14 20:36:34] [ERROR] - submission null-h01-XXX-XXX-submission.jar has 0 warnings and 1 errors!
[22-May-14 20:36:34] [ERROR] - Transformation failed :: Timeout after 1000ms

--- a/core/src/main/kotlin/org/sourcegrade/jagr/core/executor/TimeoutHandler.kt
+++ b/core/src/main/kotlin/org/sourcegrade/jagr/core/executor/TimeoutHandler.kt
@@ -46,7 +46,7 @@ object TimeoutHandler {
     private val timeoutIndividual by lazy { config.executor.timeoutIndividual }
     private val timeoutTotal by lazy { config.executor.timeoutTotal }
 
-    private val lastTimeout = ThreadLocal.withInitial { AtomicLong() }
+    private val lastTimeout = ThreadLocal.withInitial { AtomicLong(-1) }
     private val testClassNames = ThreadLocal.withInitial<List<String>> { emptyList() }
     private val mxBean = ManagementFactory.getThreadMXBean()
 

--- a/core/src/main/kotlin/org/sourcegrade/jagr/core/testing/JavaRuntimeTester.kt
+++ b/core/src/main/kotlin/org/sourcegrade/jagr/core/testing/JavaRuntimeTester.kt
@@ -81,6 +81,7 @@ class JavaRuntimeTester @Inject constructor(
             val summaryListener = SummaryGeneratingListener()
             val statusListener = TestStatusListenerImpl(logger)
             TimeoutHandler.setClassNames(map { it.className })
+            TimeoutHandler.resetTimeout()
             launcher.execute(testPlan, summaryListener, statusListener)
             // if total timeout has been reached, reset so that the rubric provider doesn't throw an error
             TimeoutHandler.disableTimeout()

